app/assets/javascripts/common/tags.js

```diff
@@ -22,10 +22,11 @@
    * @param {Array[Tag]} tags
    * @param {Array[number]} current
    * @param {Object} divs
+   * @param {Boolean|Undefined} alwaysEdit
    * @return {Tags}
    *
    */
-  t.init = function(tags, current, endpoint, divs){
+  t.init = function(tags, current, endpoint, divs, alwaysEdit ){
```

Comment:
concur with ziggy that this args list is getting long.

at some point, worth considering:

* grouping into a params object
* grouping `tags` and `currents` into `tags` (with all and current fields)

relatedly, i've been itching to call this whole state object `STATE` instead of tags, since it has more than tags in it...